### PR TITLE
Distinguish components from external add-ons

### LINKED ISSUE #2536: Split "Components" section into "Core Components" and "External Components"
While it's awesome we now have all components living under one section, this has caused users to become confused about what is actually part of Kubeflow, and what is external.

### My proposal is to split "Components" into:
* `Core Components` (all components owned by a Kubeflow WG)
* `External Components` (all other components we want to highlight)

### Proposed Structure:
```
Core Components:
- Central Dashboard
   - ... same ...
   - + Multi-Tenancy [moved]
- Notebooks [rename from "Notebook Servers"]
   - ... same ...
- Pipelines [rename from "Kubeflow Pipelines"]
   - ... same ...
- Metadata
   - ... same ...
- Katib
   - ... same ...
- KFServing
   - ... same ...
- Training Operators [rename from "Frameworks for Training"]
   - ... same ...
- Fairing
   - ... same ...
```
```
External Components:
- Istio
   - ... same as before ...
- Tools for Serving
   - ... same as before ...
- Feature Store
   - ... same as before ...
```

### PR DESCRIPTION
**Pull request by shannonbradshaw**
This fixes #2536 I'm expecting to make small fixes depending on reviews.

### CODE REVIEW

**Comment by shannonbradshaw** [comment]
cc: @RFMVasconcelos @cvenets

**Comment by thesuperzapper** [comment]
@shannonbradshaw, looks good but I think we need to update `content/en/_redirects` to prevent dead links.

@RFMVasconcelos any other thoughts, or is this good to merge (after those redirects), so we can close #2536

File: content/en/docs/components/training/_index.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 +++
-title = "Training Operators"
+title = "Frameworks for Training"
```
Comment (RFMVasconcelos) [inline content/en/docs/components/training/_index.md:2]: We made this change recently in the opposite direction. I believe that `Training operators` is more accurate.

**Review by RFMVasconcelos — COMMENTED** [review]
Thank you for this work @shannonbradshaw! 

A couple of missing things:
1. redirects in `_redirects` file so that external links are not broken

File: content/en/docs/components/training/_index.md (hunk 1 above)
Comment (shannonbradshaw) [reply content/en/docs/components/training/_index.md:2]: I'll fix it.

**Comment by shannonbradshaw** [comment]
> Thank you for this work @shannonbradshaw!
> 
> A couple of missing things:
> 
> 1. Redirects in `_redirects` file so that external links are not broken
> 2. Do a search for internal links and update them accordingly not to break them upon these change
> 
> Do you think you could take on this extra work?

Yes. I'm checking through to do redirects.

**Comment by shannonbradshaw** [comment]
@RFMVasconcelos @thesuperzapper, thanks for reviewing! I believe I've addressed your comments, including redirects and internal links. I did test and I believe the redirects and changes I made are working; however, given the tedious nature of these changes it's possible I missed something. Please let me know if you notice any further changes that need to be made.

**Comment by thesuperzapper** [comment]
@shannonbradshaw can you resolve the merge conflict? 

@RFMVasconcelos I think this is a pretty good first pass, once the merge conflict is resolved, I think you can `/approve`

**Comment by shannonbradshaw** [comment]
I broke this PR while attempting to rebase. Replacing this with a clean PR: #2630

**Pull request closed without merging**